pkg/storage/export/export/export.go

```diff
@@ -508,11 +647,93 @@ func (ctrl *VMExportController) handleIsVmSnapshot(vmExport *exportv1.VirtualMac
 	return 0, ctrl.updateVMExporVMSnapshotStatus(vmExport, pvcs, pod, service, false)
 }
 
+func (ctrl *VMExportController) handleIsVM(vmExport *exportv1.VirtualMachineExport, service *corev1.Service) (time.Duration, error) {
+	var pvcs []*corev1.PersistentVolumeClaim
+
+	inUse := false
+	vmi, exists, err := ctrl.getVmi(vmExport.Namespace, vmExport.Spec.Source.Name)
+	if err != nil {
+		return 0, err
+	}
+	if exists {
+		inUse = true
+	}
+
+	pod, exists, err := ctrl.getExporterPod(vmExport)
+	if err != nil {
+		return 0, err
+	}
+	pvcs, err = ctrl.getPVCsFromVM(vmExport.Namespace, vmExport.Spec.Source.Name)
+	if err != nil {
+		return 0, err
+	}
+	if !exists {
+		if !inUse {
+			pod, err = ctrl.createPodAndSecret(vmExport, pvcs)
```

Comment:
I think it is problematic to combine these two operations.  What if creating the pod succeeds but creating the secret fails?  Next time through the reconcile you will see the pod exists and will not attempt to recreate the secret.  This applies to all sources, not just VM, of course